OpenRA.Test/OpenRA.Game/SpatiallyPartitionedTest.cs

```diff
@@ -0,0 +1,92 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Drawing;
+using NUnit.Framework;
+using OpenRA.Primitives;
+
+namespace OpenRA.Test.OpenRA.Game
+{
+	[TestFixture]
+	class SpatiallyPartitionedTest
+	{
+		[TestCase(TestName = "SpatiallyPartitioned.At works")]
+		public void SpatiallyPartitionedAtTest()
+		{
+			var partition = new SpatiallyPartitioned<object>(5, 5, 2);
+
+			var a = new object();
+			partition.Add(a, new Rectangle(0, 0, 1, 1));
+			CollectionAssert.Contains(partition.At(new int2(0, 0)), a, "a is not present after add");
+			CollectionAssert.DoesNotContain(partition.At(new int2(0, 1)), a, "a is present in the wrong location");
+			CollectionAssert.DoesNotContain(partition.At(new int2(1, 0)), a, "a is present in the wrong location");
+
+			var b = new object();
+			partition.Add(b, new Rectangle(1, 1, 2, 2));
+			CollectionAssert.DoesNotContain(partition.At(new int2(0, 1)), b, "b is present in the wrong location");
+			CollectionAssert.DoesNotContain(partition.At(new int2(1, 0)), b, "b is present in the wrong location");
+			CollectionAssert.Contains(partition.At(new int2(1, 1)), b, "b is not present after add");
+			CollectionAssert.Contains(partition.At(new int2(2, 2)), b, "b is not present after add");
+			CollectionAssert.DoesNotContain(partition.At(new int2(2, 3)), b, "b is present in the wrong location");
+			CollectionAssert.DoesNotContain(partition.At(new int2(3, 2)), b, "b is present in the wrong location");
+			CollectionAssert.DoesNotContain(partition.At(new int2(3, 3)), b, "b is present in the wrong location");
+
+			partition.Update(b, new Rectangle(4, 4, 1, 1));
+			CollectionAssert.Contains(partition.At(new int2(0, 0)), a, "a was messed up when b updated");
```

Comment:
"a wrongly changed location when b was updated" - I'll probably amend it to that.